### ISSUE TITLE
[feature request] include top for models

### ISSUE THREAD

**Issue by Nachwa** [pytorch/vision#426]
Any plans to add (include top) option while loading models in torchvision?

**Comment by fmassa** [comment]
Could you expand on your question?
I think the current best way of having the output of different layers of your model can be found in https://github.com/pytorch/vision/pull/310#issuecomment-373055230

**Comment by Nachwa** [comment]
I thought about an easy way to load the model without its classifier. as in keras applications
model = VGG16(weights='imagenet', include_top=False). 

Example:

`
class VGG(nn.Module):

    def __init__(self, features, num_classes=1000, init_weights=True, include_top=False):
        super(VGG, self).__init__()
        self.features = features
        self.include_top = include_top
        self.classifier = nn.Sequential(
            nn.Linear(512 * 7 * 7, 4096),
            nn.ReLU(True),
            nn.Dropout(),
            nn.Linear(4096, 4096),
            nn.ReLU(True),
            nn.Dropout(),
            nn.Linear(4096, num_classes),
        )
        if init_weights:
            self._initialize_weights()

    def forward(self, x):
        x = self.features(x)
        x = x.view(x.size(0), -1)
        if self.include_top: x = self.classifier(x)
        return x
`

**Comment by fmassa** [comment]
What if the user want to remove only the final classifier layer, but not the whole `self.classifier` part?
In your snippet, you can obtain the same result just by doing `model.features(x).view(x.size(0), -1)`.
I think we might want to advertise subclassing the model to remove / add layers that you want. I think this will be the cleanest and most extensible solution.

**Comment by varunagrawal** [comment]
@Nachwa it seems like you wish for Keras like behavior in PyTorch rather than doing things the PyTorch way. I agree with @fmassa of just subclassing the model to achieve your desired behavior.

I vote to close this.

**Comment by fmassa** [comment]
@varunagrawal I'm actually thinking about ways of making finetuning / getting arbitrary elements of the models possible. Not sure if I'll come up with a good solution for this, but one possibility is by tracing the model. This gives us a full computation graph (which is static), on which we could potentially query arbitrary activations.

**Comment by datumbox** [comment]
This issue appears to be stale, so I'm closing it. If you think it should remain open, feel free to reopen it.
